shared/js/gzip/gzip_worker.js

```diff
@@ -0,0 +1,23 @@
+/* global Zlib */
+'use strict';
+
+importScripts('gzip.min.js');
+
+self.onmessage = function(message) {
+  var cmd = message.data.cmd;
+
+  if (cmd === 'gzip') {
+
+    // gzip function requires Uint8Array
+    var textEncoder = new TextEncoder('utf8');
+    var payload = message.data.payload;
+    var uint8Data = textEncoder.encode(payload);
+
+    var gzip = new Zlib.Gzip(uint8Data);
+    var compressed = gzip.compress();
+    postMessage(compressed);
+  } else {
+    console.warn('GzipWorker, unsupported cmd:', cmd);
```

Comment:
Should you be passing back errors if the command isn't 'gzip' or for some reason the compress() function fails, like low memory? There are probably useful errors that could be bubbled up to the ATPing that would allow it to fallback in case of an error.